src/hnswinsert.c

```diff
@@ -135,7 +135,7 @@ WriteNewElementPages(Relation index, HnswElement e, int m, BlockNumber insertPag
 	etupSize = HNSW_ELEMENT_TUPLE_SIZE(dimensions);
 	ntupSize = HNSW_NEIGHBOR_TUPLE_SIZE(e->level, m);
 	combinedSize = etupSize + ntupSize + sizeof(ItemIdData);
-	maxSize = BLCKSZ - MAXALIGN(SizeOfPageHeaderData) - MAXALIGN(sizeof(HnswPageOpaqueData));
+	maxSize = BLCKSZ - MAXALIGN(SizeOfPageHeaderData) - MAXALIGN(sizeof(HnswPageOpaqueData)) - sizeof(ItemIdData);
 	minCombinedSize = etupSize + HNSW_NEIGHBOR_TUPLE_SIZE(0, m) + sizeof(ItemIdData);
```

Comment:
Hmm, so 'etupSize' and 'ntupSize' don't include the size of the line pointer, but  'combinedSize' includes the size of _one_ line pointer. Seems a bit weird.

Reply:
Hi @hlinnaka   
Yes , you are correct, and here the `combinedSize` is a little difficult to understand .  
`etup` and `ntup` are separately stored in two tuples, `etupSize` or `ntupSize` is the length of the data part.  In HNSW index, it is better to store the `etup` and `ntup` on the same page, so use the `combinedSize`  which includes the line pointer of `ntup` to check if a page has enough free space for both two tuples.  
```
if (PageGetFreeSpace(page) < etupSize || (combinedSize <= maxSize && PageGetFreeSpace(page) < combinedSize))
```